### DISCUSSION TITLE
Default style for rendering PMTiles

### DISCUSSION THREAD

**Discussion by giswqs** [protomaps/PMTiles#303] [Q&A] [answered]
The [PMTiles Viewer](https://protomaps.github.io/PMTiles) can render any PMTiles with a nice-looking style. I am trying to replicate this with Python. How does PMTiles determine the layer geometry type and set the default style? For this sample dataset, the metadata does not contain the layer geometry type. I wondered how PMTiles Viewer does this under the hood to retrieve the geometry type and set the default style.  

![image](https://github.com/protomaps/PMTiles/assets/5016453/029e29f7-8014-4d2d-a188-0d4e18aaea53)

![image](https://github.com/protomaps/PMTiles/assets/5016453/0f01563d-1856-4d59-a9fe-67c500057e5f)

**Comment by bdon** [comment, marked as answer]
It does not determine a layer geometry type for each layer, because a layer can contain multiple types. it creates 3 display layers for each data layer, one for points, one for line strings, one for polygons.

**Reply by giswqs** [reply]
Good to know. Thanks for sharing. How about the colors?

**Reply by bdon** [reply]
They are generated using `d3-scale-chromatic` https://github.com/protomaps/PMTiles/blob/main/app/src/MaplibreMap.tsx#L9 but I wouldn't recommend copying those colors directly as I plan to make some improvements soon
